cli/internal/run/real_run.go

```diff
@@ -580,3 +602,16 @@ func (ec *execContext) exec(ctx gocontext.Context, packageTask *nodes.PackageTas
 	progressLogger.Debug("done", "status", "complete", "duration", taskExecutionSummary.Duration)
 	return taskExecutionSummary, nil
 }
+
+// TaskError wraps an error encountered running the given task
+type TaskError struct {
+	cause         error
+	taskIDDisplay string
+}
+
+// Unwrap allows for interoperation with standard library error wrapping
+func (te *TaskError) Unwrap() error { return te.cause }
+
+func (te *TaskError) Error() string {
+	return fmt.Sprintf("Task %v finished with error: %v", te.taskIDDisplay, te.cause)
```

Comment:
Not super strong opinions, but I feel like this produces longer than necessary loglines e.g.
`Task app-a#builderror finished with error: command .* npm run builderror exited`. IMO `app-a#builderror: command .* npm run builderror exited` gives enough context

Reply:
Good idea. Shortened